### PR TITLE
feat: add logout confirmation dialog

### LINKED ISSUE #72: Add dialog before logout
**Is your feature request related to a problem? Please describe.**
The logout button logs out people immediately.

**Describe the solution you'd like**
A confirmation dialog should be shown.

### PR DESCRIPTION
**Pull request by MarcelGarus**
Fixes: #72

### CODE REVIEW

**Comment by MarcelGarus** [comment]
Here's a screenshot of the logout dialog:
image

File: lib/l10n/intl_de.arb [hunk 1]
```diff
@@ -28,6 +28,21 @@
   "@app_errorScreen_unknown_showStackTrace": {
     "type": "text"
   },
+  "app_logOut_confirm": "Abmelden",
+  "@app_logOut_confirm": {
+    "type": "text",
+    "description": ""
+  },
+  "app_logOut_content": "Du brauchst deine Anmeldedaten, um dich wieder anzumelden.",
```
Comment (ctiedt) [inline lib/l10n/intl_de.arb]: Maybe change this something more clear like "Du musst dich danach erneut anmelden, um die App zu benutzen"

File: lib/l10n/intl_en.arb [hunk 2]
```diff
@@ -28,6 +28,21 @@
   "@app_errorScreen_unknown_showStackTrace": {
     "type": "text"
   },
+  "app_logOut_confirm": "Log out",
+  "@app_logOut_confirm": {
+    "type": "text",
+    "description": ""
+  },
+  "app_logOut_content": "You'll need your login credentials to log in again.",
```
Comment (ctiedt) [inline lib/l10n/intl_en.arb]: Don't use contractions in a formal context. Maybe change this to "You will need to provide your login credentials to use the app again."

**Review by ctiedt — DISMISSED** [review]

File: lib/l10n/intl_de.arb (hunk 1 above)
Comment (MarcelGarus) [reply lib/l10n/intl_de.arb]: Sounds good

File: lib/l10n/intl_en.arb (hunk 2 above)
Comment (MarcelGarus) [reply lib/l10n/intl_en.arb]: Will do

File: lib/app/widgets/error_widgets.dart [hunk 3]
```diff
@@ -41,7 +41,7 @@ class _MessageAndActions {
       message = s.app_errorScreen_authError;
       actions.add(SecondaryButton(
         onPressed: () => logOut(context),
-        child: Text(s.app_errorScreen_authError_logOut),
+        child: Text(s.app_errorScreen_authError),
```
Comment (JonasWanke) [inline lib/app/widgets/error_widgets.dart]: This text seems to be quite long for a simple button… Did you mean general_signOut?

File: lib/login/utils.dart [hunk 4]
```diff
@@ -1,16 +1,43 @@
...
+        title: Text(context.s.app_signOut_title),
+        content: Text(context.s.app_signOut_content),
+        actions: <Widget>[
+          FlatButton(
+            onPressed: () => Navigator.of(context).pop(false),
+            child: Text(context.s.general_cancel),
+          ),
+          FlatButton(
+            onPressed: () => Navigator.of(context).pop(true),
+            child: Text(context.s.general_signOut),
```
Comment (JonasWanke) [inline lib/login/utils.dart]: Maybe cache `s`?

File: lib/login/utils.dart [hunk 5]
```diff
@@ -1,16 +1,43 @@
...
-    builder: (_) => LoginScreen(),
-  )));
+  if (confirmed) {
+    // Actually log out.
+
+    // This should probably be awaited, but right now awaiting it
+    // leads to the issue that logging out becomes impossible.
+    unawaited(services.get<StorageService>().clear());
+
+    final navigator = Navigator.of(context, rootNavigator: true)
```
Comment (JonasWanke) [inline lib/login/utils.dart]: Why don't you use `context.rootNavigator` anymore?

**Review by JonasWanke — CHANGES_REQUESTED** [review]

File: lib/app/widgets/error_widgets.dart (hunk 3 above)
Comment (MarcelGarus) [reply lib/app/widgets/error_widgets.dart]: Yes, I meant
```suggestion
        child: Text(s.general_signOut),
```

**Review by JonasWanke — APPROVED** [review]

**Pull request merged**